### PR TITLE
Fixed the rest of the javadoc breaking errors and enabled linting

### LINKED ISSUE #975: Clean up javadoc errors
When building the docs I'm currently seeing 308 errors. These should be cleaned up or set as exceptions in the maven job. Once all the errors are cleaned up I propose replacing the [skip ci] tag with a [docs] tag that does a doc build and sets up a gh-page for visual confirmation. On merge it should also update the docs so they get patched between major releases.

### PR DESCRIPTION
**Pull request by chupman**
Signed-off-by: Chris Hupman <chupman@us.ibm.com>

There are still a ton of warnings to fix, but I focused on the errors and was able to run a build last night with linting enabled. Curious to see how it affects CI times. 

helps to resolve #975 
-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there an issue associated with this PR? Is it referenced in the commit message?
- [X] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [X] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [X] Is your initial contribution a single, squashed commit?

### For code changes:
- [X] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [X] Have you ensured that format looks appropriate for the output in which it is rendered?
- [] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1688,7 +1688,7 @@
             </build>
         </profile>
 
-        <profile>
+<!--        <profile>
```
Comment (pluradj) [inline pom.xml]: I confirmed that janusgraph-core completes with doclint enabled, but we can't re-enable doclint in the top pom.xml until all of the other janusgraph-* packages have their errors fixed.

Comment (chupman) [reply pom.xml]: I removed the changes to pom.xml and added [skip ci] to prevent wasting ci cycles

**Comment by chupman** [comment]
@pluradj I was looking and it seems like we only generate javadocs on janusgraph-core. I'm pretty sure we can enable doclinting. I can build a 0.2.0 dist with linting enabled. 0.3.0 fails, but there's only a handful of errors that I would gladly fix if we can turn linting on.

**Comment by pluradj** [comment]
@chupman Sounds fine to me. FYI, better to comment on the issue rather than a merged/closed PR. I don't typically follow them after they are closed.

**Pull request merged**